### PR TITLE
AVRO-3497 Simplify conditional expression

### PR DESCRIPTION
**Pull request by KyleSchoonover**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO/) issues and references them in the PR title. For example, "AVRO-1234: My Avro PR"
  - https://issues.apache.org/jira/browse/AVRO-3497
  - Parent Issue: https://issues.apache.org/jira/browse/AVRO-3383

### Tests

- [ ] My PR adds the following unit tests:  Added coverage for equals

### Commits

- [ ] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

**Comment by KyleSchoonover** [comment]
@zcsizmadia and @martin-g

File: lang/csharp/src/apache/test/Generic/GenericEnumTests.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+            Assert.IsFalse(genericEnum.Equals((object)genericEnum2));
+        }
+
+        [Test]
+        public void TestEqualsObjectNullObject()
+        {
+            GenericEnum genericEnum = GetBaseGenericEnum();
+
+            Assert.IsFalse(genericEnum.Equals((object)null));
```
Comment (zcsizmadia) [inline lang/csharp/src/apache/test/Generic/GenericEnumTests.cs:71]: I am surprised that this passes

File: lang/csharp/src/apache/main/Generic/GenericEnum.cs [hunk 2]
```diff
@@ -70,10 +70,13 @@ public GenericEnum(EnumSchema schema, string value)
...
-            if (obj == this) return true;
-            return (obj != null && obj is GenericEnum)
-                ? Value.Equals((obj as GenericEnum).Value, System.StringComparison.Ordinal)
-                : false;
+            if (obj == this)
+            {
+                return true;
+            }
+
+            return obj.GetType() == typeof(GenericEnum)
```
[the remark this answers is not in the document]
Comment (KyleSchoonover) [reply lang/csharp/src/apache/main/Generic/GenericEnum.cs:78]: Added back

File: lang/csharp/src/apache/test/Generic/GenericEnumTests.cs (hunk 1 above)
Comment (KyleSchoonover) [reply lang/csharp/src/apache/test/Generic/GenericEnumTests.cs:71]: Fixed

File: lang/csharp/src/apache/test/Generic/GenericEnumTests.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            Assert.IsTrue(genericEnum.Equals(genericEnum2));
+        }
+
+        [Test]
+        public void TestEqualsObjectNotEqual()
+        {
+            GenericEnum genericEnum = GetBaseGenericEnum();
+            GenericEnum genericEnum2 = new GenericEnum(Schema.Parse(baseSchema) as EnumSchema, "B");
+
+            Assert.IsFalse(genericEnum.Equals((object)genericEnum2));
```
Comment (zcsizmadia) [inline lang/csharp/src/apache/test/Generic/GenericEnumTests.cs:63]: Can the casting be removed and fix the CodeQL warnings?

**Review by zcsizmadia — CHANGES_REQUESTED** [review]

File: lang/csharp/src/apache/test/Generic/GenericEnumTests.cs (hunk 3 above)
Comment (KyleSchoonover) [reply lang/csharp/src/apache/test/Generic/GenericEnumTests.cs]: Updated

**Comment by KyleSchoonover** [comment]
I will create a separate PR for the other equals checks and add appropriate tests.

**Review by zcsizmadia — APPROVED** [review]

**Review by martin-g — APPROVED** [review]

**Comment by martin-g** [comment]
Thank you, @KyleSchoonover !

**Pull request merged**